### ISSUE TITLE
Syntax highlighting wrong in listish type signature or top-level pattern matching.

### ISSUE THREAD

**Issue by navaati** [haskell/haskell-mode#55]
Hello.

When writing something like:
```haskell
bleh, bluh ∷ Blah
[bleh,bluh] = map blahfication [e,u]

blih :: Blah
```
bleh and bluh appeir in normal font-face in haskell mode, while one would expect them to appeair highlighted like blih is. The "comma-separated multiple type signature" case should definetely be highlighted. I admit that the "pattern matching at top-level" is more tricky…

**Comment by gracjan** [comment]
This case appears to be fixed along #405, by commit https://github.com/haskell/haskell-mode/commit/7ca5e34ad6ff0f5c455c02ad2d80fcbe48fef59b

**Comment by purcell** [comment]
Indeed. Thanks @gracjan.
